tutorials/plot_spatial.py

```diff
@@ -11,40 +11,215 @@
+plot_positions(position, alpha = 1, ls = '-', marker = 'x', c = 'tab:gray', markersize = 10)
+plt.title('Tracking')
+plt.xlabel('x-position')
+plt.ylabel('y-position')
+plt.legend(['coordinates'])
+
+###################################################################################################
+
+# Compute distance between start and end point
+dist_start_end = compute_distance(x_pos[0], y_pos[0], x_pos[-1], y_pos[-1])
+# Compute distance traveled at each point
+dist_traveled = compute_distances(x_pos, y_pos)
+# Compute total distance traveled
+cumulative_dist_traveled = compute_cumulative_distances(x_pos, y_pos)
+# Compute speed at each point
+bin_widths = np.array([1, 1, 1, 2, 1.5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 0.5])
+speeds = compute_speed(x_pos, y_pos, bin_widths)
+
+###################################################################################################
+
+f, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex = True)
+# plot distance traveled at each time
+plot_positions(np.append([x_pos[1:]], [dist_traveled], axis = 0), 
+               ax = ax1, alpha = 1, ls = '-', marker = 'x', c = 'tab:pink', markersize = 10)
+ax1.set_title('Distance traveled at each point')
+ax1.set_xlabel('time (t)')
+ax1.set_ylabel('distance (u)')
+
+# plot cumulative distance traveled per time
+plot_positions(np.append([x_pos[1:]], [cumulative_dist_traveled], axis = 0),
+               ax = ax2, alpha = 1, ls = '-', marker = 'x', c = 'tab:olive', markersize = 10)
+ax2.set_title('Cumulative distance traveled at each point')
+ax2.set_xlabel('time (t)')
+ax2.set_ylabel('distance (u)')
+
+# plot speed at each time point
+plot_positions(np.append([x_pos[1:]], [speeds], axis = 0),
+               ax = ax3, alpha = 1, ls = '-', marker = 'x', c = 'tab:cyan', markersize = 10)
+ax3.set_title('Speed at each point')
+ax3.set_xlabel('time (t)')
+ax3.set_ylabel('speed (u/t)')
+
+# add some padding between subplots
+f.tight_layout(pad = 0.05)
+# make figure bigger
+f.set_size_inches((15/2.54, 20/2.54))
+
+###################################################################################################
+# 2. Divide position in spatial bin edges and plot
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Compute x- and y- spatial bin edges and plot spatial grid. 
+#
+
+###################################################################################################
+
+x_edges, y_edges = compute_spatial_bin_edges(position, bins)
+
+# Compute the width of each spatial bin
+x_bins_spatial_width = get_bin_width(x_edges)
+y_bins_spatial_width = get_bin_width(y_edges)
+print(f'The x spatial bins have width = {x_bins_spatial_width}')
+print(f'The y spatial bins have width = {y_bins_spatial_width}')
 
 ###################################################################################################
 
-# Settings
-...
+# Plot grid of spatial bins with tracking on top
+track = plot_positions(position, x_bins=x_edges, y_bins=y_edges,
```

Comment:
The `spiketools` functions don't return anything from plot functions, so I don't think you want / need `track = ` here (I presume it ends up catching `None` anyways).